demo-installer.sh

```
@@ -0,0 +1,218 @@
+        echo
+        print_color $CYAN "Enter numbers (e.g., 1,3,5) or 'all' for all, 'clear' to clear, 'back' to return:"
+        
+        read -p "Selection: " selection
+        
+        case $selection in
+            "back"|"b")
+                return 0
+                ;;
+            "clear"|"c")
+                for app in "${apps[@]}"; do
+                    SELECTED_APPS=($(printf '%s\n' "${SELECTED_APPS[@]}" | grep -v "^$app$"))
```

Shellcheck (0.11.0)

[warning] Prefer mapfile or read -a to split command output (or quote to avoid splitting).

(SC2207)